src/greedycd.jl

```diff
@@ -125,7 +128,16 @@ function _update_GreedyCD!(upd::GreedyCDUpd{T}, s::GreedyCDUpd_State{T}, X,
 
     p_init = convert(T, -1.0)
     for i in 1:n_samples
-        qi = argmax(D[i, :])
+        # qi = argmax(D[i, :])
```

Comment:
Is the idea that you're reimplementing `argmax` in a more memory-friendly way? You should be able to just use `argmax(view(D, i, :))` to avoid allocating a new slice.